bankcraft/model.py

```diff
@@ -32,11 +40,22 @@ def step(self):
 
 
     def run(self, steps):
+        # start model from previous state
+        try:
+            self.datacollector = DataCollector.from_dataframe(pd.read_csv("model_state.csv"))
+        except:
+            pass
+
         for i in range(steps):
             self.step()
 
+        # collect model state
+        self.datacollector.collect(self)
+        # append model state in the existing file
+        self.datacollector.get_model_vars_dataframe().to_csv("model_state.csv", mode='a', header=False)
         agent_money = self.datacollector.get_agent_vars_dataframe()
         return (agent_money)
+    
```

Comment:
My ignorance here -- but is it necessary to save the intermediate state to a csv? I was naively expecting that we could call run() in succession and the model would just pick up where it left off. 

Part of my question is whether get_agent_vars_dataframe() returns all data from the beginning of the simulation, or only from the last time this was called?

Reply:
This would save all the information from each step from beginning to end.  This is necessary to get the output data in the format that we want it to be.